src/parser.js

```diff
@@ -77,7 +77,18 @@ function parseWithBabylon(text) {
       "functionSent",
       "dynamicImport"
     ]
-  });
+  };
+
+  try {
+    return babylon.parse(text, babylonOptions);
+  } catch (originalError) {
+    babylonOptions.strictMode = false;
```

Comment:
Yeah... I was undecided about that when I opened the PR. Fixed in https://github.com/prettier/prettier/pull/1587/commits/128538f7467a5da1d76e6400cb01c48f700b08f9